src/files-breadcrumbs.tsx

```diff
@@ -49,19 +49,47 @@ function useHostname() {
 
 // eslint-disable-next-line max-len
 export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: string[], showHidden: boolean, setShowHidden: React.Dispatch<React.SetStateAction<boolean>>}) {
+    const [editMode, setEditMode] = React.useState(false);
+    const [newPath, setNewPath] = React.useState(path.join("/"));
```

Comment:
This looked very confusing to me at first sight, as `path` is a property, and now there's a `newPath`, which conceptually collides and may lead to inconsistent state. Seeing more of the patch I saw that this should be (and is) *just* the state of the controlled edit input. so perhaps `editPath` or so, or at least comment what it does? Also, initializing it that way feels redundant -- it would change its value every time the path changes, even when not editing it. Sure, React coalesces multiple state updates, but it's still "meh". Can it be initialized when entering edit mode?

Reply:
I can rename it to `editPath` and init it when "needed" sure.